### DISCUSSION TITLE
Caller Ghosting...

### DISCUSSION THREAD

**Discussion by scotthibbs** [w7sst/MorseRunner#104] [General]
I'm totally new to contesting so forgive me if this is basic.. But I've had this question for a long time and when watching a video review of 1.81, he notes the calls will ghost (leave or non-respond) if calls are given back incorrectly. 

He states "More realistic, just aggravating". Is this realistic in contests? Would not callers want to correct a minimum amount of time? Do they realistically ghost or send nil or tu before leaving?

Seems pretty aggravating to me to be working on a call and have them ghost me as well. But is this real life? Especially when a caller knows they have a horrible call with what seems like10 letters and slashes lol?

**Comment by w7sst** [comment]
Did he have the `LIDS` checkbox checked? There is code for ghosts that will randomly appear. I have not studied this behavior. We can look into this more if needed. Please watch this portion of the review video again to see if he was `LIDS` enabled or not. I have noticed where when I call back to a station they do not respond. This might be the ghosting behavior he is talking about.

Perhaps we should disable ghosting in Single Call mode. I assume he was running in Pile Up mode. Once we understand this better, we can convert this Discussion into an Issue.

**Comment by scotthibbs** [comment]
He does not have LIDS checked (nothing checked to get clear sounds). He's working the CQ WW contest. He also notes "If you miss them one time, they leave." He's working at 40wpm. Slowing the video down to .25 (slow motion) He received NE4EA and he sent NE and received DE NE4EA, he sent NE2EU which got ghosting. He then sent NIL and still ghosting. I don't know if this is normal behavior on the air. For someone to just leave the frequency. Especially if getting a ? or NIL shouldn't they try again? Slower?

Maybe at 40 wpm there is little patience.. I tested this at 10wpm and received an AA8 call and sent a AA3 and he responded so I sent the AA8 again and he responded with the rest two more times before ghosting me. So I'm guessing this is a random number "patience" given to the callers. Or at each incorrect they got a chance to quit. (Do they send NIL before leaving or 73 in real life?)  

Ghosting doesn't happen as often in single call mode... At least I don't remember experiencing it. Maybe this is turned off for that.

**Comment by w7sst** [comment]
We need to try this same testing with 1.68 and see what happens.

**Comment by scotthibbs** [comment]
Will do. I don't think it is different, but I'll check and let you know. 
I sent an email to Jay W4ZXT to ask about if and how ghosting occurs in real world contesting.

**Comment by scotthibbs** [comment]
I do get ghosting in 1.67. I butchered a PY call right away and they left immediately. Others stood by as I butchered their calls, and some left after several tries. One sent me an R after several tries knowing I didn't get it right. (contest mode with lids automatically checked) I am at 15wpm and I really do think this is phenomenon might occur more often at faster speeds.  

Jay W4ZXT responded and stated "I’ve only had a few ghost away. Most of the time people hang with you until you get their call. I guess Morse runner 1.68 was more accurate in that regard. I was always taught that if they want the points , then they’ll stick with it." 

So his impression is that ours does ghost faster. Maybe our version, instead of ghosting,  we have that caller send 5 wpm slower or Farnsworth it. I'm guessing this is what would happen in real life, especially NE4EA that Jay missed.  

I'm thinking we leave this here and when we get down the road and several contests in with more feedback we can revisit to see if this is a feature we want to change.

**Comment by w7sst** [comment]
I have noticed changes in the implementation of partial callsign matching
code. Perhaps we are seeing a change in a partial callsign match. When we
work on this, we can explore differences in this function.

When someone works on this, take a look at all the other author branches
(any branch with the suffix `*-normalized`).

**Reply by K6OK** [reply]
I'm a contester and I think the program as is mirrors the real world quite well.  Occasionally on the air when I send a station's call I get dead air and I have a call them a 2nd time (this is exclusive of doubling).  This happens in contesting and it happens in the program too, so that's good.  Also, on the air, if I spend too much time with a station due to repeats, mistakes, etc. some of the other callers give up and leave.  The program does this too ( I think ) as I've had to CQ to refresh the pileup with new callers.  So, right now, I'd vote for no changes to the program.

**Reply by tekenny** [reply]
I agree

**Comment by scotthibbs** [comment]
Got an email from W4ZXT (he also posted a 1.82 video but no voiceover just him using the program). Here's what he said:

"One thing I’ve noticed is that when running naqp they give call , name and state. After they repeat about 3 times they go away. Usually I run qrm qrn and Qsb. Most real contests they give their call either until you get it or 7-8 times before they give up. Haha. My only complaint about 1.81 and 1.82. Tnx brother. Hope you had a great Christmas!"

He's brought up ghosting before and there is no change between the original or our program versions. I'm wondering how we handle this. I know we got a couple votes to leave ghosting alone above, but I was thinking we might make this a user option. It would make since that ghosting behavior would be more contest specific. For example, CWOPS (which is focused on teaching and more cw activity) may not ghost and NAQP which I don't believe to be very competitive may actually try 7 or 8 more times than very competitive competitions where ghosting my happen regularly. Thus I think ghosting is operator and contest specific behavior. 

The user option would mean we have a checkmark for "normal ghosting" in the "band conditions" section. Or probably duplicating "Activity" with "Ghosting" with the number for Ghosting where 0 is no ghosting and the number represents how many tries before ghosting. Best yet, we need both, a checkmark for normal ghosting, and if not selected, the number input isn't greyed out and selectable per above. 

(Also, "band conditions" needs to be renamed to just "conditions" imho.)

Hope you had a very Merry Christmas, Scott

edit: restating for clarity: I suggest that we duplicate the "Activity" number selection to be a checkbox for "normal ghosting" that when not selected would have a number to control ghosting with "0" being no ghosting and a number to 9 for how many retries before ghosting.
